modin/core/dataframe/pandas/dataframe/protocol/dataframe.py

```diff
@@ -0,0 +1,1223 @@
+        """
+        return Column(
+            self._df.mask(row_positions=None, col_labels=[name]),
+            allow_copy=self._allow_copy,
+            offset=self._offset,
+        )
+
+    def get_columns(self) -> Iterable[Column]:
+        """
+        Return an iterator yielding the columns.
+
+        Yields
+        ------
+        Column
+            The ``Column`` object(s).
+        """
+        for name in self._df.columns:
+            yield Column(
+                self._df.mask(row_positions=None, col_labels=[name]),
+                allow_copy=self._allow_copy,
+                offset=self._offset,
+            )
+
+    def select_columns(self, indices: Sequence[int]) -> "DataFrame":
+        """
+        Create a new DataFrame by selecting a subset of columns by index.
+
+        names : Sequence[int]
+            Column indices to be selected out of the DataFrame.
+
+        Returns
+        -------
+        DataFrame
+            A new DataFrame with selected a subset of columns by index.
+        """
+        if not isinstance(indices, collections.Sequence):
+            raise ValueError("`indices` is not a sequence")
+
+        return DataFrame(
+            self._df.mask(row_positions=None, col_positions=indices),
+            allow_copy=self._allow_copy,
+            offset=self._offset,
+        )
+
+    def select_columns_by_name(self, names: Sequence[str]) -> "DataFrame":
+        """
+        Create a new DataFrame by selecting a subset of columns by name.
+
+        Parameters
+        ----------
+        names : Sequence[str]
+            Column names to be selected out of the DataFrame.
+
+        Returns
+        -------
+        DataFrame
+            A new DataFrame with selected a subset of columns by name.
+        """
+        if not isinstance(names, collections.Sequence):
+            raise ValueError("`names` is not a sequence")
+
+        return DataFrame(
+            self._df.mask(row_positions=None, col_labels=names),
+            allow_copy=self._allow_copy,
+            offset=self._offset,
+        )
+
+    def get_chunks(self, n_chunks: Optional[int] = None) -> Iterable["DataFrame"]:
+        """
+        Return an iterator yielding the chunks.
+
+        By default `n_chunks=None`, yields the chunks that the data is stored as by the producer.
+        If given, `n_chunks` must be a multiple of `self.num_chunks()`,
+        meaning the producer must subdivide each chunk before yielding it.
+
+        Parameters
+        ----------
+        n_chunks : int, optional
+            Number of chunks to yield.
+
+        Yields
+        ------
+        DataFrame
+            A ``DataFrame`` object(s).
+        """
+        offset = 0
+        if n_chunks is None:
+            for length in self._row_lengths:
+                yield DataFrame(
+                    self._df.mask(
+                        row_positions=list(range(length)), col_positions=None
```

Comment:
```suggestion
                        row_positions=range(length), col_positions=None
```
processing slice-like indexers is generally faster (and allows to do zero-copy masking for arrow tables in OmniSci backend).

Reply:
fixed